src/To.Be.Generated/AssistantsPageEnumerator.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace OpenAI.Assistants;
+
+internal partial class AssistantsPageEnumerator : PageEnumerator<Assistant>
```

Comment:
Is the plan for `To.be.Generated` types to be added to the TypeSpec specification, or do these already exist in the TypeSpec and this is just a generator gap?"

Reply:
TypeSpec has a proposed design for pagination, but it is still under development.  I've been working with the team to ensure that the SCM types can be used for anything the TypeSpec can represent, and last I checked, they are looking into unifying the 3p pagination design with the 3p LRO design, at least at lower levels, around the observation that the "get next page" method for pagination is similar conceptually to the "get status update" method for LROs.

To answer your question - to my knowledge, the generator is currently unable to implement generation of these types due to the TypeSpec pieces not being available yet.